packages/amplify-graphql-transformer-core/src/types/utils.ts

```diff
@@ -0,0 +1,229 @@
+  FunctionDirectiveStack: 'function',
+  PredictionsDirectiveStack: 'predictions',
+  SearchableStack: 'openSearch',
+};
+
+const rootStackNameInConstruct = 'transformer-root-stack';
+
+/**
+ * Test Cases
+// ['rootStack' , 'Name*'] id = resource
+// ['rootStack' , 'scope'] id = Name
+// ['rootStack' , 'NestedStack' , 'Name*'] id  = resource
+// ['rootStack' , 'NestedStack'] id = Name
+ */
+
+export const getStackMeta = (constructPathArr: string[], id: string, nestedStackArr: string[], node: Construct): ConstructResourceMeta => {
+  const resource = node as CfnResource;
+  if (nestedStackArr.find(val => val === constructPathArr[1])) {
+    const nestedStackName = nestedStackArr.find(val => val === constructPathArr[1]);
+    const resourceName = constructPathArr.filter(path => path !== nestedStackName && path !== rootStackNameInConstruct).join('');
+    return {
+      resourceName,
+      resourceType: resource.cfnResourceType,
+      nestedStack: {
+        stackName: nestedStackName!,
+        stackType: stacksTypes[nestedStackName!] ?? stacksTypes['MODELS'],
+      },
+    };
+  } else {
+    // root stack
+    const resourceName = constructPathArr.filter(path => path !== rootStackNameInConstruct).join('');
+    return {
+      resourceName: id === 'Resource' ? resourceName : `${resourceName}${id}`,
+      resourceType: resource.cfnResourceType,
+      rootStack: {
+        stackName: constructPathArr[0],
+        stackType: stacksTypes.API,
+      },
+    };
+  }
+};
+
+export const cast = (resource: CfnResource) => {
+  if (resource.cfnResourceType === 'AWS::DynamoDB::Table') {
+    return resource as CfnTable;
+  } else if (resource.cfnResourceType === 'AWS::AppSync::GraphQLApi') {
+    return resource as CfnGraphQLApi;
+  } else if (resource.cfnResourceType === 'AWS::AppSync::ApiKey') {
+    return resource as CfnApiKey;
+  } else if (resource.cfnResourceType === 'AWS::AppSync::GraphQLSchema') {
+    return resource as CfnGraphQLSchema;
+  } else if (resource.cfnResourceType === 'AWS::AppSync::DataSource') {
+    return resource as CfnDataSource;
+  } else if (resource.cfnResourceType === 'AWS::IAM::Role') {
+    return resource as CfnRole;
+  } else if (resource.cfnResourceType === 'AWS::IAM::Policy') {
+    return resource as CfnPolicy;
+  } else if (resource.cfnResourceType === 'AWS::AppSync::Resolver') {
+    return resource as CfnResolver;
+  } else if (resource.cfnResourceType === 'AWS::AppSync::FunctionConfiguration') {
+    return resource as CfnFunctionConfiguration;
+  } else if (resource.cfnResourceType === 'AWS::Lambda::Function') {
+    return resource as CfnFunction;
+  } else if (resource.cfnResourceType === 'AWS::Lambda::EventSourceMapping') {
+    return resource as CfnEventSourceMapping;
+  } else if (resource.cfnResourceType === 'AWS::Elasticsearch::Domain') {
+    return resource;
+  } else {
+    return undefined;
+  }
+};
+
+export const convertToAppsyncResourceObj = (amplifyObj: any) => {
```

Comment:
nit: Could use `$TSObject` instead of `any`. Here and in the functions below

Reply:
I have used any since this package is not dependent on amplify-cli-core